ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/writer/delta/DeltaStreamWriter.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.ingestor.implementation.writer.delta
+
+import io.delta.tables.DeltaTable
+import org.apache.commons.configuration2.Configuration
+import org.apache.commons.lang3.StringUtils
+import org.apache.spark.sql.catalyst.expressions.objects.AssertNotNull
+import org.apache.spark.sql.functions.{col, lit, when}
+import org.apache.spark.sql.streaming.{OutputMode, StreamingQuery, Trigger}
+import org.apache.spark.sql.{Column, DataFrame, Row, functions}
+import org.slf4j.LoggerFactory
+import za.co.absa.hyperdrive.ingestor.api.utils.{ConfigUtils, StreamWriterUtil}
+import za.co.absa.hyperdrive.ingestor.api.writer.{StreamWriter, StreamWriterFactory, StreamWriterProperties}
+
+private[writer] class DeltaStreamWriter(destination: String,
+                                        trigger: Trigger,
+                                        checkpointLocation: String,
+                                        partitionColumns: Seq[String],
+                                        keyColumn: String,
+                                        opColumn: String,
+                                        deletedValue: String,
+                                        sortColumns: Seq[String],
+                                        sortColumnsCustomOrder: Map[String, Seq[String]],
+                                        val extraConfOptions: Map[String, String]) extends StreamWriter {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+  if (StringUtils.isBlank(destination)) {
+    throw new IllegalArgumentException(s"Invalid DELTA destination: '$destination'")
+  }
+
+  override def write(dataFrame: DataFrame): StreamingQuery = {
+    dataFrame.writeStream
+      .trigger(trigger)
+      .outputMode(OutputMode.Append())
+      .option(StreamWriterProperties.CheckpointLocation, checkpointLocation)
+      .options(extraConfOptions)
+      .foreachBatch((df: DataFrame, batchId: Long) => {
+        if(!DeltaTable.isDeltaTable(df.sparkSession, destination)) {
```

Comment:
If it's not a delta table, but some data is present, I think the program should fail, otherwise we might delete data